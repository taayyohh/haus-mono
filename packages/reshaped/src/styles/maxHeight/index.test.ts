import getMaxHeightStyles from "./index";

describe("Styles/MaxHeight", () => {
	test("handles px value", () => {
		expect(getMaxHeightStyles("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(getMaxHeightStyles("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(getMaxHeightStyles(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getMaxHeightStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getMaxHeightStyles({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
