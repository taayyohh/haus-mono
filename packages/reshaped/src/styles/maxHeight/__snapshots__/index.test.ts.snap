// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/MaxHeight > handles % value 1`] = `
{
  "classNames": [
    "_literal_7989d5",
  ],
  "variables": {
    "--rs-max-h-s": "50%",
  },
}
`;

exports[`Styles/MaxHeight > handles px value 1`] = `
{
  "classNames": [
    "_literal_7989d5",
  ],
  "variables": {
    "--rs-max-h-s": "50px",
  },
}
`;

exports[`Styles/MaxHeight > handles responsive value 1`] = `
{
  "classNames": [
    "_literal_7989d5",
    "_literal--l_7989d5",
  ],
  "variables": {
    "--rs-max-h-l": "50%",
    "--rs-max-h-s": "50px",
  },
}
`;

exports[`Styles/MaxHeight > handles undefined value 1`] = `null`;

exports[`Styles/MaxHeight > handles unit value 1`] = `
{
  "classNames": [
    "_unit_7989d5",
  ],
  "variables": {
    "--rs-max-h-s": 5,
  },
}
`;
